src/mopidy/internal/jsonrpc.py

```diff
@@ -1,61 +1,80 @@
 import inspect
-import json
 import traceback
 from collections.abc import Callable
-from typing import Any, Literal, TypeAlias, TypedDict, TypeVar
+from typing import Any, Literal, Self, TypeAlias, TypeVar
 
+import msgspec
 import pykka
 
-T = TypeVar("T")
+from mopidy import models
 
+T = TypeVar("T")
 
-class JsonRpcNotification(TypedDict):
+RequestId: TypeAlias = str | int | float
+Model: TypeAlias = (
+    models.Album
+    | models.Artist
+    | models.Image
+    | models.Playlist
+    | models.Ref
+    | models.SearchResult
+    | models.TlTrack
+    | models.Track
+)
+ParamValue: TypeAlias = None | bool | int | float | str | Model
+Param: TypeAlias = ParamValue | list[ParamValue]
+
+
+class Request(msgspec.Struct, kw_only=True, omit_defaults=True):
     jsonrpc: Literal["2.0"]
     method: str
-    params: list[Any] | dict[str, Any]
-
-
-JsonRpcRequestId: TypeAlias = str | int | float
-
+    params: list[Param] | dict[str, Param] | None = None
```

Comment:
```suggestion
        params: list[Param] | dict[str, Any] | None = None
```
Workaround for methods that have a `dict` parameter:
* https://docs.mopidy.com/latest/api/core/#mopidy.core.TracklistController.filter
* https://docs.mopidy.com/latest/api/core/#mopidy.core.LibraryController.search
* https://docs.mopidy.com/latest/api/core/#mopidy.core.LibraryController.get_distinct